edge/pkg/edged/volume_host.go

```diff
@@ -138,8 +137,18 @@ func (evh *edgedVolumeHost) GetConfigMapFunc() func(namespace, name string) (*ap
 func (evh *edgedVolumeHost) GetExec(pluginName string) utilexec.Interface  { return nil }
 func (evh *edgedVolumeHost) GetHostIP() (net.IP, error)                    { return nil, nil }
 func (evh *edgedVolumeHost) GetNodeAllocatable() (api.ResourceList, error) { return nil, nil }
-func (evh *edgedVolumeHost) GetNodeLabels() (map[string]string, error)     { return nil, nil }
-func (evh *edgedVolumeHost) GetNodeName() types.NodeName                   { return types.NodeName(evh.edge.nodeName) }
+func (evh *edgedVolumeHost) GetNodeLabels() (map[string]string, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		klog.Errorf("couldn't determine hostname: %v", err)
+		return nil, err
+	}
+	if len(evh.edge.nodeName) != 0 {
+		hostname = evh.edge.nodeName
+	}
+	return map[string]string{api.LabelHostname: hostname}, nil
```

Comment:
thanks for the fixing, how about using `evh.edge.initialNode()`, it will return all the node info, also the labels.

```
node, err := evh.edge.initialNode()
if err != nil {
	return nil, fmt.Errorf("error retrieving node: %v", err)
}
return node.Labels, nil
```

Reply:
thanks,get it.Before it was rude.

Reply:
Yes... In future we can get node info from the local cache. And could you please squash the commits?

Reply:
ok.